src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -69,8 +69,10 @@ public function copy($originFile, $targetFile, $override = false)
                 throw new IOException(sprintf('Failed to copy "%s" to "%s".', $originFile, $targetFile), 0, null, $originFile);
             }
 
-            if (stream_is_local($originFile) && $bytesCopied !== filesize($originFile)) {
-                throw new IOException(sprintf('Failed to copy the whole content of "%s" to "%s %g bytes copied".', $originFile, $targetFile, $bytesCopied), 0, null, $originFile);
+            $bytesOrigin = filesize($originFile);
+
+            if (stream_is_local($originFile) && $bytesCopied !== $bytesOrigin) {
+                throw new IOException(sprintf('Failed to copy the whole content of "%s" to "%s" (%g of %g bytes copied).', $originFile, $targetFile, $bytesCopied, $bytesOrigin), 0, null, $originFile);
```

Comment:
Nice change :+1: 
Why not move the declaration of `$bytesOrigin` inside of the `if`-check/statement itself? That way it is only declarated if the `stream_is_local` check is true.